include/eve/detail/function/simd/x86/slide_right.hpp

```diff
@@ -0,0 +1,160 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/implementation.hpp>
+
+namespace eve::detail
+{
+  template<real_scalar_value T, typename N, x86_abi ABI, std::ptrdiff_t Shift>
+  EVE_FORCEINLINE wide<T,N,ABI> slide_right_ ( EVE_SUPPORTS(sse2_)
+                                            , wide<T,N,ABI> v, index_t<Shift>
+                                            ) noexcept
+  requires(Shift <= N::value)
+  {
+          if constexpr(Shift == 0)        return v;
+    else  if constexpr(Shift == N::value) return wide<T,N,ABI>{0};
+    else
+    {
+      if constexpr( std::same_as<ABI,x86_128_>)
+      {
+        constexpr auto shift = Shift*sizeof(T);
+        using i_t = as_integer_t<wide<T,N,ABI>, unsigned>;
+
+        auto const b  = bit_cast(v, as_<i_t>());
+        return bit_cast(i_t(_mm_bslli_si128( b, shift)), as(v));
+      }
+      else if constexpr( std::same_as<ABI,x86_256_>)
+      {
+        if constexpr( current_api >= avx2)
+        {
+          using i_t = as_integer_t<wide<T,N,ABI>>;
+          constexpr auto offset = Shift * sizeof(T);
+
+          i_t vi  = bit_cast(v, as_<i_t>{});
+          i_t bvi = _mm256_permute2x128_si256(vi, vi, 0x08);
+
+          if constexpr(offset == 16)
+          {
+            return bit_cast(bvi,as(v));
+          }
+          else  if constexpr(offset <  16)
+          {
+            return bit_cast(i_t(_mm256_alignr_epi8 (vi, bvi, 16 - offset)) , as(v));
+          }
+          else
+          {
+            return bit_cast(i_t(_mm256_bslli_epi128(bvi, offset - 16)), as(v));
+          }
```

Comment:
Maybe format these in one line each like for shift_left?

Reply:
Sure